src/Components/BookNowModal/index.js

```diff
@@ -0,0 +1,159 @@
+    const [errorMessageEmail, setErrorMessageEmail] = useState('');
+    const [errorMessageAdult, setErrorMessageAdult] = useState('');
+    const [errorMessageChildren, setErrorMessageChildren] = useState('');
+    const [inputName, setinputName] = useState('')
+    const [inputEmail, setinputEmail] = useState('')
+    const [inputAdult, setinputinputAdult] = useState('')
+    const [inputChildren, setinputChildren] = useState('')
+
+    const handleName = (e) => {
+        setinputName(e.target.value)
+    }
+
+    const handleEmail = (e) => {
+        setinputEmail(e.target.value)
+    }
+
+    const handleAdult = (e) => {
+        setinputinputAdult(e.target.value)
+    }
+    const handleChildren = (e) => {
+        setinputChildren(e.target.value)
+    }
+
+    const handleSuccess = async (e) => {
+        let isValid = true;
+
+        if (inputName.length === 0) {
+            e.preventDefault()
+            isValid = false;
+            setErrorMessageName("This field is required!")
+        } else if (!validator.isAlpha(inputName)) {
+            e.preventDefault()
+            isValid = false;
+            setErrorMessageName("It must be a valid name!")
+        } else {
+
+            setErrorMessageName(" ")
+        }
+
+        if (inputEmail.length === 0) {
+            e.preventDefault()
+            isValid = false;
+            setErrorMessageEmail("This field is required!")
+        } else if (!validator.isEmail(inputEmail)) {
+            e.preventDefault()
+            isValid = false;
+            setErrorMessageEmail("It should be a valid email!")
+        } else {
+            setErrorMessageEmail(" ")
+        }
+
+        if (inputAdult.length === 0) {
+            e.preventDefault()
+            isValid = false;
+            setErrorMessageAdult("This field is required!")
+        } else {
+            setErrorMessageAdult(" ")
+        }
+
+        if (inputChildren.length === 0) {
+            e.preventDefault()
+            isValid = false;
+            setErrorMessageChildren("This field is required!")
+        } else {
+            setErrorMessageChildren(" ")
+        }
+
+        if (isValid) {
+            const rawResponse = await fetch('http://127.0.0.1:3000/hotels/' + props.hotelId, {
+                method: 'POST',
+                headers: {'Content-Type': 'application/json'},
+                body: JSON.stringify({
+                    name: inputName,
+                    email: inputEmail,
+                    checkin: props.checkin,
+                    checkout: props.checkout,
+                    adult_guests: inputAdult,
+                    child_guests: inputChildren
+                })
+            });
+            const response = await rawResponse.json()
+            if (!response.success) {
+                setErrorMessageRespons('Something went wrong, please try again!')
+            } else {
+                history.push('/success')
+            }
```

Comment:
i would swap these around, do the `history.push` in the if with the error as the else